explorer/data/prelude.carbon

```diff
@@ -86,6 +86,137 @@ impl String as EqWith(Self) {
   }
 }
 
+interface LessWith(U:! Type) {
+  fn Less[me: Self](other: U) -> Bool;
+  // TODO: NotEqual with default impl
```

Comment:
These NotEqual comments should only be on the Equal interface (generally true through your edits)

Context: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/comparison_operators.md#equality